### DISCUSSION TITLE
Response sent before the files are actually available

### DISCUSSION THREAD

**Discussion by Faber-smythe** [expressjs/multer#1138] [General]
Hi. I didn't feel like opening an issue for this, I was mostly wondering if I understand things right.

I have a setup with Nuxt and Express, and I am using multer.array() to upload two files at once. And basically the app needs to be notified when the upload is done in order to use the new images, but the callback is fired before the files are available.

From what I gathered, multer is part of the router chain, so in the below code, the post handler is supposedly fired once the file are done uploading.

```
router.post("/upload", upload.array("files"), (req, res)=>{
  // files are now available
  res.status(201).json({
    log: 'upload is complete', 
    files: req.files
  })
});
```
Back in the app, I can receive the response and log the file object just fine, something like this.
```
{ 
  destination : "./uploads",
  encoding : "7bit",
  fieldname : "files",
  filename : "my_image.jpg",
  mimetype : "image/jpeg",
  originalname : "my image.jpg",
  path : "assets/textures/my_image.jpg",
  size : 2297865
}
```
So the object is available. But when I want to use the filename to fetch the newly uploaded image, it throws an error because the file is not available yet. I came to this conclusion because when I use setTimeout for about 4000ms it solves the issue.

Does this problem sound familiar to anybody ?

**Comment by UnKnoWn-Consortium** [comment]
I think you can do a `fs.stat` before `res.status(201).json(…)` to see if the files are properly saved and determine if it is really a server-side issue.
